jina/peapods/runtimes/zmq/zed.py

```diff
@@ -204,26 +204,39 @@ def _handle(self) -> 'ZEDRuntime':
             Handle does not handle explicitly message because it may wait for different messages when different parts are expected
         :return: ZEDRuntime procedure.
         """
-
-        if (
-            not re.match(self.envelope.header.target_peapod, self.name)
-            or self.request_type != 'DataRequest'
-        ):
+        # skip executor for non-DataRequest
+        if self.request_type != 'DataRequest':
+            self.logger.debug(f'skip executor: not data request')
             return self
 
         # migrated from the previously RouteDriver logic
+        # set dealer id
         if self._idle_dealer_ids:
             dealer_id = self._idle_dealer_ids.pop()
             self.envelope.receiver_id = dealer_id
 
             # when no available dealer, pause the pollin from upstream
             if not self._idle_dealer_ids:
                 self._zmqstreamlet.pause_pollin()
+            self.logger.debug(
+                f'using route, set receiver_id: {self.envelope.receiver_id}'
+            )
 
+        # skip executor if target_peapod mismatch
+        if not re.match(self.envelope.header.target_peapod, self.name):
```

Comment:
equality check may be more efficient than regex match

Reply:
We only want to compare the beginning of the string. Regex seems slightly better.


```python
import random
import string
import re

pat = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
foo_str = [pat + '/ZEDRuntime'] * 10000 + ['foo/ZEDRuntime'] * 10000

from datetime import datetime

beg = datetime.now()
for foo in foo_str:
    result = re.match(pat, foo)
end = datetime.now()
print(f're cost: {(beg-end).microseconds} microsecond')

beg = datetime.now()
for foo in foo_str:
    result = foo.startswith(pat)
end = datetime.now()
print(f'string cost: {(beg-end).microseconds} microsecond')

# re cost: 980502 microsecond
# string cost: 996321 microsecond
```

Reply:
hmmm, using regex match actually introduces a potential bug. I've switched to the plain string matching

Reply:
- where is "We only want to compare the beginning of the string" come from?
- please revert back to `re.match`

Reply:
> where is "We only want to compare the beginning of the string" come from?
This is what the old codes have implemented. I've added a test to show the issue. Check it out at https://github.com/jina-ai/jina/blob/72f3a6db6bfac2d5ca7efbfb6d721fcbf493946d/tests/integration/v2_api/test_func_routing.py#L191

Reply:
reverted